editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas.spec.browser2.tsx

```diff
@@ -0,0 +1,316 @@
+          ${positionLine}
+          left: 50,
+          top: ${50 + elementTopShifts[0]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'red',
+        }}
+        data-label='Red'
+      />
+      <div
+        data-uid='div-green'
+        data-testid='div-green'
+        style={{
+          ${positionLine}
+          left: 60,
+          top: ${60 + elementTopShifts[1]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'green',
+        }}
+        data-label='Green'
+      />
+      <div
+        data-uid='div-blue'
+        data-testid='div-blue'
+        style={{
+          ${positionLine}
+          left: 70,
+          top: ${70 + elementTopShifts[2]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'blue',
+        }}
+        data-label='Blue'
+      />
+    </div>
+  );
+};
+
+export var storyboard = (
+  <Storyboard data-uid="storyboard">
+    <Scene
+      data-uid="scene-1"
+      style={{ position: "absolute", left: 0, top: 0, width: 375, height: 812 }}
+    >
+      <App data-uid="app" />
+    </Scene>
+  </Storyboard>
+);
+`
+}
+
+function createAndRenderModifiedProject(modifiedFiles: {
+  [filename: string]: string
+}): Promise<EditorRenderResult> {
+  const project = createModifiedProject(modifiedFiles)
+  return renderTestEditorWithModel(project, 'await-first-dom-report')
+}
+
+function createExampleProject(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+): Promise<EditorRenderResult> {
+  return createAndRenderModifiedProject({
+    [StoryboardFilePath]: exampleProject(positionSetting, elementTopShifts),
+  })
+}
+
+async function checkOverlappingElements(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+  focusedOn: ElementPath | null,
+  targetDivId: string,
+  cmdPressed: boolean,
+  x: number,
+  y: number,
+  expectedSelectedElements: Array<ElementPath>,
+): Promise<void> {
+  const renderResult = await createExampleProject(positionSetting, elementTopShifts)
+
+  const targetDiv = renderResult.renderedDOM.getByTestId(targetDivId)
+  const targetDivBounds = targetDiv.getBoundingClientRect()
+
+  const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+  if (focusedOn != null) {
+    await act(async () => {
+      const dispatchDone = renderResult.getDispatchFollowUpActionsFinished()
+      await renderResult.dispatch([setFocusedElement(focusedOn)], true)
+      await dispatchDone
+    })
+  }
+
+  await act(async () => {
+    const dispatchDone = renderResult.getDispatchFollowUpActionsFinished()
+    fireEvent(
+      canvasControlsLayer,
+      new MouseEvent('mousedown', {
+        detail: 1,
+        bubbles: true,
+        cancelable: true,
+        metaKey: cmdPressed,
+        clientX: targetDivBounds.left + x,
+        clientY: targetDivBounds.top + y,
+        buttons: 1,
+      }),
+    )
+    await dispatchDone
+  })
```

Comment:
with the new synchonously rendering canvas, you can use the synchronous `act` function (which takes a synchronous callback function)

this is how the other new tests simulate clicks, no need to await stuff!
```suggestion
  act(() => {
    fireEvent(
      canvasControlsLayer,
      new MouseEvent('mousedown', {
        detail: 1,
        bubbles: true,
        cancelable: true,
        metaKey: cmdPressed,
        clientX: targetDivBounds.left + x,
        clientY: targetDivBounds.top + y,
        buttons: 1,
      }),
    )
  })
```